### DISCUSSION TITLE
Exclude path(s) from project folder

### DISCUSSION THREAD

**Discussion by thestumbler** [modm-io/modm#1073] [Q&A]
I see there is a provision in the generated SConstruct file,
```
sources += env.FindSourceFiles(".", ignorePaths=ignored)
```

And in the syntax for the LBUILD file:
```
    # You can also copy entire folders
    env.copy("folder/", dest="renamed/")
    # and ignore specific RELATIVE files/folders
    env.copy("folder/", ignore=env.ignore_files("*.txt", "this_specific_file.hpp"))
    # or ignore specific ABSOLUTE paths
    env.copy("folder/", ignore=env.ignore_paths("*/folder/*.txt"))
```

Can I set an ignore path(s) in the `project.xml` file?  For example, can I stash some files in `src/playground/` and have them ignored by the build? Or do I have to modify SConstruct / Makefile manually?

**Comment by salkinium** [comment]
>  Or do I have to modify SConstruct / Makefile manually?

Yes. The top-level files are explicitly minimal and BSD-licensed so that you can modify them exactly as you wish.
We made that choice because the lbuild configuration of modm would get incredibly bloated if it actually covered every configuration option of all the build systems we support.

**Reply by salkinium** [reply]
It's also not that great to have to regenerate all the code with lbuild all the time just to update one path string. Modifying the build system directly will probably be a better choice in terms of usability and probably also in correctness (assuming the build system is smart enough to understand such config changes).

**Reply by thestumbler** [reply]
Got it. In practice I'm now sure how often one needs to run LBUILD, once you get everything setup. In my case, doing a bunch of up-front tests, brining and and out different peripheral modules, jumping from one board to another, etc., I'm running it quite frequently. But that's probably a odd case.
